Lesson16/Mesut/arraysAndObjectsManipulations.js

```diff
@@ -4,72 +4,164 @@
 // Input: {name: "John", age: 30, city: "New York"}
 // Output: ["name", "age", "city"]
+debugger;
+function justKeys(obj){
+    let keys=[];
+    for(let i=0;i<Object.values(obj).length;i++){
+        keys.push(Object.values(obj)[i]);
 
-console.log('----------------------------');
-// 8. Write a function that takes an object and returns the values as an array.
-// Input: {name: "John", age: 30, city: "New York"}
-// Output: ["John", 30, "New York"]
+    }
+    console.log(keys);
+}
 
-// array is an object {0: "arrayValue1", 1: "arrayValue2", 2: "arrayValue2"};
-// ["arrayValue1", "arrayValue2", "arrayValue2"];
-// primitive values are number, string, boolean
-// other values (e.g. array, function, object) are objects
+justKeys({name: "John", age: 30, city: "New York"});
 
-console.log('----------------------------');
+console.log('------------exercise 8----------------');
+console.log('-------------exercise 9---------------');
 // 9. Write a function that takes an object and a key, and returns true if the key exists in the object, otherwise false.
 // Input: ({name: "John", age: 30, city: "New York"}, "age")
 // Output: true
 // Input: ({name: "John", age: 30, city: "New York"}, "surname")
 // Output: false
 
-console.log('----------------------------');
+// function check(obj,key){
+//     for(let i=0;i<Object.values(obj).length;i++){
+//         if()
+//     }
+
+
+
+
+
+console.log('--------------exercise 10--------------');
 // 10. Write a function that takes an object and a key-value pair, and adds the key-value pair to the object.
 // Input: ({name: "John", age: 30}, "city", "New York")
 // Output: {name: "John", age: 30, city: "New York"}
 
-console.log('----------------------------');
+function addKeyValue(obj,key,value){
+    let x ={};
+    obj[key]=value;
+   return obj;
+}
+console.log(addKeyValue({name: "John", age: 30}, "city", "New York"));
+
+console.log('-------------exercise 11---------------');
 // 11. Write a function that takes an object and a key, and removes the key-value pair from the object.
 // Input: ({name: "John", age: 30, city: "New York"}, "city")
 // Output: {name: "John", age: 30}
 
-console.log('----------------------------');
+function deleteKeyValue(obj, key){
+    delete obj[key];
+    return obj;
+    
+}
+
+console.log(deleteKeyValue({name: "John", age: 30, city: "New York"}, "city"));
+
+console.log('-------------exercise 12---------------');
 // 12. Write a function that takes two arrays of equal length and returns an object with the first array as keys and the second array as values.
 // Input: (["name", "age", "city"], ["John", 30, "New York"])
 // Output: {name: "John", age: 30, city: "New York"}
+//?????????????????????????????????????????????????????/
+function concatenateArrays(arr1, arr2){
+    let obj={};
+    for(let i=0;i<arr1.length;i++){
+        obj[arr1[i]]=arr2[i];   
+   
+    }
+   return obj;
+}
+
+console.log(concatenateArrays=(["name", "age", "city"], ["John", 30, "New York"]));
```

Comment:
This change should solve the issue with this exercise 
```suggestion
console.log(concatenateArrays(["name", "age", "city"], ["John", 30, "New York"]));
```